pkg/cmd/run.go

```diff
@@ -48,12 +48,14 @@ func NewRunMicroshiftCommand() *cobra.Command {
 
 func RunMicroshift(cfg *config.MicroshiftConfig, flags *pflag.FlagSet) error {
 	if err := cfg.ReadAndValidate(flags); err != nil {
-		logrus.Fatal(err)
+		klog.ErrorS(err, "Error in reading and validating flags")
+		os.Exit(1)
 	}
 
 	// fail early if we don't have enough privileges
 	if config.StringInList("node", cfg.Roles) && os.Geteuid() > 0 {
-		logrus.Fatalf("Microshift must be run privileged for role 'node'")
+		klog.InfoS("Microshift must be run privileged for role 'node'", "privilege", cfg.Roles, "user id", os.Geteuid())
```

Comment:
```suggestion
		klog.ErrorS(nil, "Microshift must be run privileged for role 'node'", "privilege", cfg.Roles, "user id", os.Geteuid())
```

I have double checked that ErrorS can be called with nil err (inspected the klog source code and tried it out): 
```
[majopela@fedora check]$ go run main.go 
E0929 11:16:42.612793  722761 main.go:6] "errors-test"
E0929 11:16:42.612852  722761 main.go:7] errorf-test

[majopela@fedora check]$ cat main.go 
package main

import "k8s.io/klog/v2"

func main() {
	klog.ErrorS(nil, "errors-test")
	klog.Errorf("errorf-test")
}
```

Reply:
I read this about the calling [ErrorS with nil](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#using-errors) 
That is why I replaced 
```
Calling ErrorS with nil as error is semi-acceptable if there is error condition that deserves a stack trace at this origin point. For expected errors (errors that can happen during routine operations) please consider using klog.InfoS and pass error in err key instead.
```

Reply:
In this case it's a Fatalf, which is not a routine case, just a final crash.